src/data/finsupp/basic.lean

```diff
@@ -774,8 +775,97 @@ end
 @[simp] lemma graph_eq_empty {f : α →₀ M} : f.graph = ∅ ↔ f = 0 :=
 (graph_injective α M).eq_iff' graph_zero
 
+/-- When turned into a list of key/value pairs, a graph has no duplicate keys. -/
+lemma graph_nodupkeys (f : α →₀ M) : (list.map prod.to_sigma f.graph.to_list).nodupkeys :=
+begin
+  rw [list.nodupkeys, list.keys, list.map_map, prod.fst_to_sigma_comp, list.nodup_map_iff_inj_on],
+  { rintros ⟨b, m⟩ hb ⟨c, n⟩ hc (rfl : b = c),
+    rw [mem_to_list, finsupp.mem_graph_iff] at hb hc,
+    dsimp at hb hc,
+    rw [←hc.1, hb.1] },
+  { apply nodup_to_list }
+end
+
 end graph
 
+end finsupp
+
+/-! ### Declarations about `list.lookup_finsupp` and `alist.lookup_finsupp` -/
+
+section lookup_finsupp
+
+variable [has_zero M]
+
+namespace alist
+open list
+
+/-- Converts an association list into a finitely supported function via `alist.lookup`, sending
+absent keys to zero. -/
+@[simps] def lookup_finsupp (l : alist (λ _ : α, M)) : α →₀ M :=
+{ support := (l.1.filter $ λ x, sigma.snd x ≠ 0).keys.to_finset,
+  to_fun := λ a, (l.lookup a).get_or_else 0,
+  mem_support_to_fun := λ a, begin
+    simp_rw [mem_to_finset, list.mem_keys, list.mem_filter, ←mem_lookup_iff],
+    cases lookup a l;
+    simp
+  end }
+
+alias lookup_finsupp_to_fun ← lookup_finsupp_apply
+
+end alist
+
+namespace list
+
+/-- Converts a list of key/value pairs into a finitely supported function via `list.lookup`, sending
+absent keys to zero. -/
+def lookup_finsupp (l : list (Σ x : α, M)) : α →₀ M := l.to_alist.lookup_finsupp
+
+@[simp] lemma to_alist_lookup_finsupp (l : list (Σ x : α, M)) :
+  l.to_alist.lookup_finsupp = l.lookup_finsupp :=
+rfl
+
+@[simp] lemma lookup_finsupp_support (l : list (Σ x : α, M)) :
```

Comment:
I don't think we have a convention but it looks like `support_` (86 usages in this file) is more popular than `_support` (18 usages).
```suggestion
@[simp] lemma support_lookup_finsupp (l : list (Σ x : α, M)) :
```